src/apps/weblib/js-api/plugins.js

```diff
@@ -0,0 +1,149 @@
+export default class Plugins {
+  constructor(parent, plugins) {
+    this._parent = parent
+    this._plugins = []
+    if (plugins) {
+      if (!Array.isArray(plugins)) throw new Error('Plugins must be an array')
+      plugins.forEach((plugin) => this.register(plugin, true))
+    }
+  }
+
+  getName(instance) {
+    const plugin = this._plugins.find((plugin) => plugin.instance === instance)
+    return plugin?.name
+  }
+
+  register(instance, enabled) {
+    this._validate(instance)
+    const name = instance.meta.name
+    if (instance.register) {
+      instance.register(this._parent)
+    }
+    const plugin = { name, instance, undefined }
```

Comment:
why do we need the undefined property here?

Reply:
We don't, supposed to be something like this, thanks.
```
    const plugin = { name, instance, event: undefined }
```